src/go/rpk/pkg/cli/acl/acl.go

```diff
@@ -83,6 +83,21 @@ in the --allow-principal and --deny-principal flags. This command will add the
 "User:" prefix for you if it is missing. The wildcard '*' matches any user.
 Creating an ACL with user '*' grants or denies the permission for all users.
 
+ROLES
+
+Alternatively, an ACL may be bound to a role. To create a role-bound ACL,
+follow the same procedure used to create a user ACL, but use the --allow-role
+and --deny-role flags to specify the role(s) to which the ACL should apply.
+
+You should be aware of a few key differences when creating a role ACL:
+
+  * Wildcard matching role names is not allowed. So --allow-role '*' is illegal.
+  * The type prefix is not necessary. The flag fully specifies that the
```

Comment:
what happens if the user passes a type? is it ignored?

Reply:
effectively, yes, unless it matches a specific prefix:

If the user passes, e.g., "User:my_role" we bind the resulting ACL to "RedpandaRole:User:my_role". This is analogous to the behavior of `--allow-principal` and friends.

However, if the user passes "RedpandaRole:my_role" we bind the ACL to "RedpandaRole:my_role". This is _also_ analogous to `--allow-principal` but the prefix itself is deliberately left undocumented, since a role is always a role.